test/widgets/register_page_test.dart

```diff
@@ -0,0 +1,322 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/util/buttons/loading_button.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:motis_mitfahr_app/welcome/pages/after_registration_page.dart';
+import 'package:motis_mitfahr_app/welcome/pages/register_page.dart';
+import 'package:progress_state_button/progress_button.dart';
+
+import '../util/mocks/mock_server.dart';
+import '../util/mocks/request_processor.dart';
+import '../util/mocks/request_processor.mocks.dart';
+import '../util/pump_material.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+  const String username = 'motismitfahrapp';
+  const String email = 'motismitfahrapp@gmail.com';
+  const String password = 'abc&EFG0';
+
+  setUpAll(() async {
+    MockServer.setProcessor(processor);
+    SupabaseManager.setCurrentProfile(null);
+  });
+
+  group('RegisterPage', () {
+    final Finder usernameFieldFinder = find.byKey(const Key('registerUsernameField'));
+    final Finder emailFieldFinder = find.descendant(
+      of: find.byKey(const Key('registerEmailField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder passwordFieldFinder = find.descendant(
+      of: find.byKey(const Key('registerPasswordField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder passwordConfirmFieldFinder = find.descendant(
+      of: find.byKey(const Key('registerPasswordConfirmField')),
+      matching: find.byType(TextFormField),
+    );
+    final Finder registerButtonFinder = find.byType(LoadingButton);
+
+    testWidgets('Shows the register page', (WidgetTester tester) async {
+      await pumpMaterial(tester, const RegisterPage());
+
+      final Finder registerFormFinder = find.byType(RegisterForm);
+      final RegisterFormState formState = tester.state(registerFormFinder);
+
+      expect(formState.buttonState, ButtonState.idle);
+    });
+
+    testWidgets('Validates the email', (WidgetTester tester) async {
```

Comment:
can we refactor the test of the emailField since the code is the same for register and login, same for Password field.

Reply:
I know, I thought the same thing... But I don't actually know how you would do that. I can try to look into it. Probably just move those tests to EmailField and then mock it some way

Reply:
Wrote unit tests and removed the "Validates the email" test cases. It's not 100% the same thing, but should be close enough.